### PR TITLE
Gh 2461/fix transaction type

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #2461 

Changes proposed in this pull request:
- Remove mock response
- Enhance decoded data display 
- Use estimated nonce& safeTxGas instead of safe once

### CODE REVIEW

File: Multisig/Data/Services/Safe Client Gateway Service/TransactionPreviewRequest.swift [hunk 1]
```diff
@@ -59,8 +59,7 @@ extension SafeClientGatewayService {
         chainId: String,
         completion: @escaping (Result<TransactionPreviewRequest.ResponseType, Error>) -> Void) -> URLSessionTask? {
 
-        // Will always return the mocked response in completion
-        asyncExecuteMock(request: TransactionPreviewRequest(safe: transaction.safe!,
```
Comment (DmitryBespalov) [inline Multisig/Data/Services/Safe Client Gateway Service/TransactionPreviewRequest.swift:63]: you can delete the mock data (files), I think

File: Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/WCIncomingTransactionRequestViewController.swift [hunk 2]
```diff
@@ -147,5 +188,16 @@ class WCIncomingTransactionRequestViewController: ReviewSafeTransactionViewContr
         default: return super.tableView(tableView, cellForRowAt: indexPath)
         }
     }
+
+    override func onSuccess(transaction: SCGModels.TransactionDetails) {
+        DispatchQueue.main.async { [unowned self] in
```
Comment (DmitryBespalov) [inline Multisig/UI/WalletConnect/Server/Safes/Incoming Transaction/WCIncomingTransactionRequestViewController.swift]: I'd use `weak` here

**Review by DmitryBespalov — APPROVED** [review]

File: Multisig/Logic/Models/Transaction.swift [hunk 3]
```diff
@@ -156,6 +156,17 @@ extension Transaction {
         updateSafeTxHash()
     }
 
+    mutating func update(nonce: UInt256String? ,safeTxGas: UInt256String?) {
```
Comment (elgatovital) [inline Multisig/Logic/Models/Transaction.swift]: space before coma

File: Multisig/Data/Services/Safe Client Gateway Service/TransactionPreviewRequest.swift (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig/Data/Services/Safe Client Gateway Service/TransactionPreviewRequest.swift:63]: I would keep it for now just for future testing if needed

**Review by elgatovital — APPROVED** [review]

**Pull request merged**